### ISSUE TITLE
Creating a new FormGroup instance for the same <form> Element does not initialize disabled states correctly

### ISSUE THREAD

**Issue by BaggersIO** [angular/angular#44478]
### Which @angular/* package(s) are the source of the bug?

forms

### Is this a regression?

No

### Description

If a form is initialized multiply times with a new FormGroup, default disabled states of the controls will not apply.

It seems, that the form states (enabled/disabled) are somehow cached in the view. So, as a workaround, I destroyed the view and created a new one. You can see the workaround in the stackblitz, too.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-ivy-6fg5hd?file=src%2Fapp%2Fapp.component.ts

### Please provide the exception or error you saw

```true
The view is not rendered correctly / form controls have wrong enabled/disabled states.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 13.1.1
Node: 14.17.2
Package Manager: npm 6.14.13
OS: darwin x64

Angular: 13.1.0
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1301.1
@angular-devkit/build-angular   13.1.1
@angular-devkit/core            13.1.1
@angular-devkit/schematics      13.1.1
@angular/cli                    13.1.1
@schematics/angular             13.1.1
rxjs                            6.6.7
typescript                      4.5.4
```

### Anything else?

The stackblitz example is built with Angular 12, but the same bug appears on a local ng13 project.

**Comment by destus90** [comment]
Seems like the work is already underway in this area

https://github.com/angular/angular/pull/43499

**Comment by AndrewKushnir** [comment]
As @destus90 mentioned above, this issue looks similar to #22556 (with a possible fix in PR #43499), so I'm closing this issue as a duplicate. Please subscribe to #22556 to get further updates. Thank you.
